src/error_and_log.js

```diff
@@ -0,0 +1,36 @@
+const isProduction = require('./is_production')
+const LoggedError = require('./logged_error')
+
+/**
+ * Create a LoggedError, log it, and return it.
+ * The log output will look like this:
+ *
+ * ``` text
+ * 2020-03-05T03:28:45.721Z 138edc1d-9a4e-49ab-9850-620489903565
+ *   Unable to reach Trello. Application is likely misconfigured.
+ *   Error: 404 - Not Found
+ *       at repl:1:86
+ *       at Script.runInThisContext (vm.js:120:20)
+ *       at REPLServer.defaultEval (repl.js:430:29)
+ *       at bound (domain.js:426:14)
+ *       at REPLServer.runBound [as eval] (domain.js:439:12)
+ * ```
+ *
+ * The first line of log output is a formatted date and unique identifier
+ * (correlation ID). The second line is the message, and the third and
+ * subsequent lines are the original error and stack trace.
+ *
+ * @param {string} message a human-readable message to be logged
+ * @param {Error} error the original error to be logged
+ * @returns {LoggedError}
+ */
+module.exports = (message, error) => {
```

Comment:
Could this function be a constructor? When we use it elsewhere, it seems out of place to be calling a function to create an error. 

Also, this will log even if the error is caught and recovered from. Is that correct? It seems like it might be better to delegate to the catcher to provide their own message in some circumstances

Reply:
Yeah, I separated the logging from the new error type. I like the latest approach much better, thanks for pointing this out.

At some point I may look for a logging framework or better way to manage logging levels so they can be managed at runtime, but don't want to go too overboard with that just yet.